From: Abdur Rehman <abdur_rehman@mentor.com>
Date: Fri, 29 Jan 2016 17:11:19 +0500
Subject: [PATCH] xserver-common: use xvfb as xserver if available

If available use Xvfb as X server. This is helpful in
situations where we don't have a physical display port as in
the case of Cyclone V.

Create screen 0 for Xvfb with WxHxD = 640x480x16

Upstream-Status: Inappropriate [configuration]

Signed-off-by: Abdur Rehman <abdur_rehman@mentor.com>
---
 etc/X11/xserver-common |    6 ++++++
 1 file changed, 6 insertions(+)

diff --git a/X11/xserver-common b/X11/xserver-common
index 4dc48c4..002134e 100755
--- a/X11/xserver-common
+++ b/X11/xserver-common
@@ -25,6 +25,9 @@ fi
 if [ -f $BINDIR/Xglamo ]; then
   XSERVER=Xglamo
 fi
+if [ -f $BINDIR/Xvfb ]; then
+  XSERVER=Xvfb
+fi
 
 if [ -f /etc/profile ]; then
   . /etc/profile
@@ -237,6 +240,9 @@ fi
 if [ "$XSERVER" != "Xorg" ] ; then
     ARGS="$ARGS $MOUSE $KDRIVEARGS"
 fi
+if [ "$XSERVER" = "Xvfb" ] ; then
+    ARGS="-screen 0 640x480x16"
+fi
 
 # let a local script modify the variables                    
 if [ -e /etc/X11/xserver-system ] ; then                    
-- 
1.7.9.5

